// @salt-ds/core/dist-cjs/card/Card.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var clsx = require('clsx');
var React = require('react');
var window = require('@salt-ds/window');
var styles = require('@salt-ds/styles');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var Card$1 = require('./Card.css.js');

const withBaseName = makePrefixer.makePrefixer("saltCard");
const Card = React.forwardRef(function Card2(props, ref) {
  const {
    className,
    disabled,
    interactable,
    children,
    variant = "primary",
    ...rest
  } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-card",
    css: Card$1,
    window: targetWindow
  });
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    className: clsx.clsx(
      withBaseName(),
      withBaseName(variant),
      {
        [withBaseName("disabled")]: disabled,
        [withBaseName("interactable")]: interactable
      },
      className
    ),
    ref,
    ...rest,
    children
  });
});

exports.Card = Card;
