// @salt-ds/core/dist-cjs/card/InteractableCard.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var clsx = require('clsx');
var React = require('react');
var window = require('@salt-ds/window');
var styles = require('@salt-ds/styles');
var capitalize = require('../utils/capitalize.js');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var useInteractableCard = require('./useInteractableCard.js');
var InteractableCard$1 = require('./InteractableCard.css.js');

const withBaseName = makePrefixer.makePrefixer("saltInteractableCard");
const InteractableCard = React.forwardRef(function InteractableCard2(props, ref) {
  const {
    accentPlacement = "bottom",
    children,
    className,
    variant = "primary",
    disabled,
    onBlur,
    onClick,
    onKeyUp,
    onKeyDown,
    ...rest
  } = props;
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-interactable-card",
    css: InteractableCard$1,
    window: targetWindow
  });
  const { active, cardProps } = useInteractableCard.useInteractableCard({
    disabled,
    onKeyUp,
    onKeyDown,
    onBlur,
    onClick
  });
  const { tabIndex, ...restCardProps } = cardProps;
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    ...restCardProps,
    className: clsx.clsx(
      withBaseName(),
      withBaseName(variant),
      withBaseName(`accent${capitalize.capitalize(accentPlacement)}`),
      {
        [withBaseName("disabled")]: disabled,
        [withBaseName("active")]: active
      },
      className
    ),
    ...rest,
    ref,
    children
  });
});

exports.InteractableCard = InteractableCard;
