// @salt-ds/core/dist-es/accordion/AccordionPanel.js
import { jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { useWindow } from '@salt-ds/window';
import { useComponentCssInjection } from '@salt-ds/styles';
import { clsx } from 'clsx';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useAccordion } from './AccordionContext.js';
import css_248z from './AccordionPanel.css.js';

const withBaseName = makePrefixer("saltAccordionPanel");
const AccordionPanel = forwardRef(
  function AccordionPanel2(props, ref) {
    const { children, className, ...rest } = props;
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-accordion-panel",
      css: css_248z,
      window: targetWindow
    });
    const { id, expanded } = useAccordion();
    return /* @__PURE__ */ jsx("div", {
      ref,
      className: clsx(withBaseName(), className),
      role: "region",
      id: `${id}-panel`,
      "aria-labelledby": `${id}-header`,
      "aria-hidden": !expanded ? "true" : void 0,
      hidden: !expanded,
      ...rest,
      children: /* @__PURE__ */ jsx("div", {
        className: withBaseName("inner"),
        children: /* @__PURE__ */ jsx("div", {
          className: withBaseName("content"),
          children
        })
      })
    });
  }
);

export { AccordionPanel };
