// @salt-ds/core/dist-es/card/InteractableCard.js
import { jsx } from 'react/jsx-runtime';
import { clsx } from 'clsx';
import { forwardRef } from 'react';
import { useWindow } from '@salt-ds/window';
import { useComponentCssInjection } from '@salt-ds/styles';
import { capitalize } from '../utils/capitalize.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useInteractableCard } from './useInteractableCard.js';
import css_248z from './InteractableCard.css.js';

const withBaseName = makePrefixer("saltInteractableCard");
const InteractableCard = forwardRef(function InteractableCard2(props, ref) {
  const {
    accentPlacement = "bottom",
    children,
    className,
    variant = "primary",
    disabled,
    onBlur,
    onClick,
    onKeyUp,
    onKeyDown,
    ...rest
  } = props;
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-interactable-card",
    css: css_248z,
    window: targetWindow
  });
  const { active, cardProps } = useInteractableCard({
    disabled,
    onKeyUp,
    onKeyDown,
    onBlur,
    onClick
  });
  const { tabIndex, ...restCardProps } = cardProps;
  return /* @__PURE__ */ jsx("div", {
    ...restCardProps,
    className: clsx(
      withBaseName(),
      withBaseName(variant),
      withBaseName(`accent${capitalize(accentPlacement)}`),
      {
        [withBaseName("disabled")]: disabled,
        [withBaseName("active")]: active
      },
      className
    ),
    ...rest,
    ref,
    children
  });
});

export { InteractableCard };
