// @salt-ds/core/dist-cjs/toast/ToastContent.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var clsx = require('clsx');
var React = require('react');
var window = require('@salt-ds/window');
var styles = require('@salt-ds/styles');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var ToastContent$1 = require('./ToastContent.css.js');

const withBaseName = makePrefixer.makePrefixer("saltToastContent");
const ToastContent = React.forwardRef(function ToastContent2({ children, className, ...restProps }, ref) {
  const targetWindow = window.useWindow();
  styles.useComponentCssInjection({
    testId: "salt-toast-content",
    css: ToastContent$1,
    window: targetWindow
  });
  return /* @__PURE__ */ jsxRuntime.jsx("div", {
    className: clsx.clsx(withBaseName(), className),
    ...restProps,
    ref,
    children
  });
});

exports.ToastContent = ToastContent;
