// @salt-ds/core/dist-es/border-item/BorderItem.js
import { jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { clsx } from 'clsx';
import { GridItem } from '../grid-item/GridItem.js';
import css_248z from './BorderItem.css.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { useWindow } from '@salt-ds/window';
import { useComponentCssInjection } from '@salt-ds/styles';

const BORDER_POSITION = [
  "north",
  "west",
  "center",
  "east",
  "south"
];
const withBaseName = makePrefixer("saltBorderItem");
const BorderItem = forwardRef(
  ({
    children,
    className,
    position,
    sticky = false,
    style,
    ...rest
  }, ref) => {
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-border-item",
      css: css_248z,
      window: targetWindow
    });
    const gridItemStyles = {
      ...style,
      "--gridItem-gridArea": position
    };
    return /* @__PURE__ */ jsx(GridItem, {
      ref,
      className: clsx(
        withBaseName(),
        "saltGridItem-area",
        {
          [withBaseName("sticky")]: sticky
        },
        className
      ),
      style: gridItemStyles,
      ...rest,
      children
    });
  }
);

export { BORDER_POSITION, BorderItem };
