// @salt-ds/core/dist-cjs/flex-item/FlexItem.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
var useResponsiveProp = require('../utils/useResponsiveProp.js');
var clsx = require('clsx');
var FlexItem$1 = require('./FlexItem.css.js');
var window = require('@salt-ds/window');
var styles = require('@salt-ds/styles');

const withBaseName = makePrefixer.makePrefixer("saltFlexItem");
const FLEX_ITEM_ALIGNMENTS = [
  "start",
  "end",
  "center",
  "stretch"
];
const FlexItem = React.forwardRef(
  ({
    as,
    align,
    children,
    className,
    shrink,
    grow,
    basis,
    style,
    ...rest
  }, ref) => {
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-flex-item",
      css: FlexItem$1,
      window: targetWindow
    });
    const Component = as || "div";
    const flexItemShrink = useResponsiveProp.useResponsiveProp(shrink, 1);
    const flexItemGrow = useResponsiveProp.useResponsiveProp(grow, 0);
    const flexItemBasis = useResponsiveProp.useResponsiveProp(basis, "auto");
    const itemStyle = {
      "--saltFlexItem-alignment": align,
      "--saltFlexItem-shrink": flexItemShrink,
      "--saltFlexItem-grow": flexItemGrow,
      "--saltFlexItem-basis": flexItemBasis,
      ...style
    };
    return /* @__PURE__ */ jsxRuntime.jsx(Component, {
      className: clsx.clsx(withBaseName(), className),
      ref,
      style: itemStyle,
      ...rest,
      children
    });
  }
);

exports.FLEX_ITEM_ALIGNMENTS = FLEX_ITEM_ALIGNMENTS;
exports.FlexItem = FlexItem;
