// @salt-ds/core/dist-es/link/Link.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { clsx } from 'clsx';
import { TearOutIcon } from '@salt-ds/icons';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { Text } from '../text/Text.js';
import '../text/Display.js';
import '../text/Headings.js';
import '../text/Label.js';
import '../text/TextNotation.js';
import '../text/TextAction.js';
import css_248z from './Link.css.js';
import { useWindow } from '@salt-ds/window';
import { useComponentCssInjection } from '@salt-ds/styles';

const withBaseName = makePrefixer("saltLink");
const Link = forwardRef(function Link2({
  IconComponent = TearOutIcon,
  href,
  className,
  children,
  variant = "primary",
  target = "_self",
  ...rest
}, ref) {
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-link",
    css: css_248z,
    window: targetWindow
  });
  return /* @__PURE__ */ jsxs(Text, {
    as: "a",
    className: clsx(withBaseName(), className),
    href,
    ref,
    target,
    variant,
    ...rest,
    children: [
      children,
      target === "_blank" && /* @__PURE__ */ jsxs(Fragment, {
        children: [
          IconComponent && /* @__PURE__ */ jsx(IconComponent, {
            className: withBaseName("icon"),
            "aria-hidden": true
          }),
          /* @__PURE__ */ jsx("span", {
            className: withBaseName("externalLinkADA"),
            children: "External"
          })
        ]
      })
    ]
  });
});

export { Link };
