// @salt-ds/core/dist-es/banner/Banner.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { clsx } from 'clsx';
import { StatusIndicator } from '../status-indicator/StatusIndicator.js';
import { useWindow } from '@salt-ds/window';
import { useComponentCssInjection } from '@salt-ds/styles';
import css_248z from './Banner.css.js';

const withBaseName = makePrefixer("saltBanner");
const Banner = forwardRef(function Banner2({ children, className, variant = "primary", status = "info", ...rest }, ref) {
  const targetWindow = useWindow();
  useComponentCssInjection({
    testId: "salt-banner",
    css: css_248z,
    window: targetWindow
  });
  return /* @__PURE__ */ jsxs("div", {
    className: clsx(
      withBaseName(),
      withBaseName(status),
      withBaseName(variant),
      className
    ),
    ref,
    ...rest,
    "aria-live": "polite",
    children: [
      /* @__PURE__ */ jsx(StatusIndicator, {
        status,
        className: withBaseName("icon")
      }),
      children
    ]
  });
});

export { Banner };
