// @salt-ds/core/dist-cjs/accordion/AccordionGroup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var clsx = require('clsx');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var window = require('@salt-ds/window');
var styles = require('@salt-ds/styles');
var AccordionGroup$1 = require('./AccordionGroup.css.js');

const withBaseName = makePrefixer.makePrefixer("saltAccordionGroup");
const AccordionGroup = React.forwardRef(
  function AccordionGroup2(props, ref) {
    const { className, ...rest } = props;
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-accordion-group",
      css: AccordionGroup$1,
      window: targetWindow
    });
    return /* @__PURE__ */ jsxRuntime.jsx("div", {
      className: clsx.clsx(withBaseName(), className),
      ref,
      ...rest
    });
  }
);

exports.AccordionGroup = AccordionGroup;
