// @salt-ds/core/dist-es/pagination/PageButton.js
import { jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { clsx } from 'clsx';
import { Button } from '../button/Button.js';
import { makePrefixer } from '../utils/makePrefixer.js';
import '../utils/useFloatingUI/useFloatingUI.js';
import '../utils/useId.js';
import '../salt-provider/SaltProvider.js';
import '../viewport/ViewportProvider.js';
import { usePaginationContext } from './usePaginationContext.js';
import { useWindow } from '@salt-ds/window';
import { useComponentCssInjection } from '@salt-ds/styles';
import css_248z from './PageButton.css.js';

const withBaseName = makePrefixer("saltPageButton");
const PageButton = forwardRef(
  function PageButton2({ page, selected, disabled }, ref) {
    const targetWindow = useWindow();
    useComponentCssInjection({
      testId: "salt-page-button",
      css: css_248z,
      window: targetWindow
    });
    const { count, onPageChange } = usePaginationContext();
    const onClick = (event) => {
      onPageChange(event, page);
    };
    return /* @__PURE__ */ jsx(Button, {
      "aria-label": `Page ${page} of ${count}`,
      "aria-current": selected ? "page" : void 0,
      variant: "secondary",
      className: clsx(withBaseName(), {
        [withBaseName("selected")]: selected,
        [withBaseName("fixed")]: page < 100
      }),
      onClick,
      disabled,
      ref,
      children: page
    });
  }
);

export { PageButton };
