// @salt-ds/core/dist-cjs/banner/BannerActions.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var clsx = require('clsx');
var makePrefixer = require('../utils/makePrefixer.js');
require('../utils/useFloatingUI/useFloatingUI.js');
require('../utils/useId.js');
require('../salt-provider/SaltProvider.js');
require('../viewport/ViewportProvider.js');
var BannerActions$1 = require('./BannerActions.css.js');
var styles = require('@salt-ds/styles');
var window = require('@salt-ds/window');

const withBaseName = makePrefixer.makePrefixer("saltBannerActions");
const BannerActions = React.forwardRef(
  function BannerActions2(props, ref) {
    const { className, ...rest } = props;
    const targetWindow = window.useWindow();
    styles.useComponentCssInjection({
      testId: "salt-banner-actions",
      css: BannerActions$1,
      window: targetWindow
    });
    return /* @__PURE__ */ jsxRuntime.jsx("div", {
      className: clsx.clsx(withBaseName(), className),
      ...rest,
      ref
    });
  }
);

exports.BannerActions = BannerActions;
